### ISSUE TITLE
support cf_reader reading multiple variables stored across multiple files

### ISSUE THREAD

**Issue by taobrienlbl** [LBL-EESA/TECA#353]
Following a discussion on May 1 among @taobrienlbl, @burlen, @elbashandy and @ndkeen , the cf_reader currently doesn't support reading multiple variables from multiple files.  The use case we are looking at is that U wind, V wind, and specific humidity are each in separate sets of files.  For now, let's assume we can use symbolic linking to put all the files in the same directory, so that the file regex doesn't yet need to be updated.

The leading idea in our discussion was to modify the (time, file) mapping to be a ((time, variable), file) mapping instead.

One thing to consider is that there may be edge cases where the file-timestep organization differs among files.  For example, the list of files below, from one of the CMIP6 runs (`/global/cfs/projectdirs/m3522/cmip6/CMIP6/CMIP/CNRM-CERFACS/CNRM-CM6-1-HR/historical/r1i1p1f2/6hrLev`), covers the same time period but has files/dates grouped differently between U wind and V wind:

```ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_200910010600-200911010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201004010600-201005010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201005010600-201006010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201008010600-201009010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201009010600-201010010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201106010600-201107010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201107010600-201108010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201112010600-201201010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201207010600-201208010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201210010600-201211010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201301010600-201302010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201308010600-201309010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201403010600-201404010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201408010600-201409010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201409010600-201410010000.nc
ua_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201412010600-201501010000.nc
```

```
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201004010600-201005010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201008010600-201009010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201012010600-201101010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201101010600-201102010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201102010600-201103010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201103010600-201104010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201109010600-201110010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201112010600-201201010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201206010600-201207010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201301010600-201302010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201303010600-201304010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201310010600-201311010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201401010600-201402010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201403010600-201404010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201407010600-201408010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201408010600-201409010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201410010600-201411010000.nc
hus_6hrLev_CNRM-CM6-1-HR_historical_r1i1p1f2_gr_201412010600-201501010000.nc
```
In our discussion, we thought that the new mapping approach described above would handle this; we'll want to make sure.

**Comment by burlen** [comment]
I see 2 issues here, and propose solving in two parts. Part 1 handles multiple folders issues, ignores multiple time resolutions in a way that would let us process that data in a limited way. Part 2 would add support for multiple time resolutions.

1. data sets w/ multiple variables stored in multiple folders
2. data sets w/ multiple variables stored in multiple folders, some or all on different time axes.

The work around for part 1 assumes that lower resolution time axes are subsets of the higher resolution axes. we could work around 2 by presenting the coarsest time axis to down stream. Then for all requests made to the reader across all variables there is a disk image that can be fetched.

The solution for 2  is complicated. Initial thoughts: extend our metadata so that time axis, in addition to being a property of the dataset, is a property of the variable. requests for a variable at a time where there is no disk image, could be handled by a time interpolation algorithm. This could potentially be an extension to the `teca_cartesian_mesh_regrid` algorithm which handles spatial interpolation across cartesian meshes. Given that many of our execution drivers parallelize over time steps some care in how time steps are presented to the pipeline is needed. We wouldn't want to processes all steps in the highest resolution, unless it was actually needed. For that reason we'd need to add some runtime controls about how the time dimension is presented.

**Comment by burlen** [comment]
@taobrienlbl 
```
~$ls /global/cfs/projectdirs/m3522/cmip6/CMIP6/CMIP/CNRM-CERFACS/CNRM-CM6-1-HR/historical/r1i1p1f2/6hrLev
ls: cannot access '/global/cfs/projectdirs/m3522/cmip6/CMIP6/CMIP/CNRM-CERFACS/CNRM-CM6-1-HR/historical/r1i1p1f2/6hrLev': Permission denied
```

**Comment by burlen** [comment]
resolved by #413
